implementations/rust/ockam/ockam_api/src/authenticator/direct.rs

```diff
@@ -190,6 +193,21 @@ impl DirectAuthenticator {
         );
         self.attributes_writer.put_attributes(id, entry).await
     }
+
+    async fn list_members(
+        &self,
+        enroller: &IdentityIdentifier,
+    ) -> Result<HashMap<IdentityIdentifier, AttributesEntry>> {
+        // TODO: move filter to `list` function
+        let all_attributes = self.attributes_reader.list().await?;
+        let attested_by_me: HashMap<IdentityIdentifier, AttributesEntry> = all_attributes
```

Comment:
Since the value must match the function's return type, you can generally elide the inner generic type hints with:
```suggestion
        let attested_by_me: HashMap<_, _> = all_attributes
```

In this instance you could even omit the type hint of HashMap altogether.
```suggestion
        let attested_by_me = all_attributes
```

Reply:
Pretty sure compiler was complaining last time I did that. It doesn't anymore though.

Reply:
Most likely once you added `Ok(attested_by_me)` it was able to successfully do inference without hinting, but couldn't tell how the value would be used up until that line was added.